### PR TITLE
Handle mailserver connection error

### LINKED ISSUE #3787: [WIP] Improve network offline and mail server error messaging
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to know when problems are preventing me from sending or receiving messages so that I can wait or take other action

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

Errors such as being offline or mail server issues prevent users from chatting. Therefore we should clearly surface these to inform the user. 

#### Expected behavior
[comment]: # (Describe what you expected to happen.)

When a user can't send or receive messages, they should know the state. The user can either wait or take action (eg, connect to wifi)

#### Actual behavior
[comment]: # (Describe what happened.)

When a user can't send or receive messages, they have no idea of the state and can get frustrated.

### Solution

Todo: designs.

### PR DESCRIPTION
**Pull request by dmitryn**
fixes #3787 

### Summary:

* Do not try to reconnect to mailserver when app is offline
* If app is online, but mailserver is disconnected, show notification to the user
* Tap on notification starts reconnecting process

### Testing notes (optional):
You could probably emulate mailserver disconnect by blocking IP 51.15.35.110 (for mainnet)

### Steps to test:
- Open Status
- Emulate mailserver disconnect
- Wait for couple of minutes for notification to appear (stay on Home screen or inside any chat)
- Tap on it to reconnect

status: ready

### CODE REVIEW

**Review by yenda — APPROVED** [review]
Nice I hope this get merged fast this is probably a good performance improvement on flaky connections that were probably spamming the mailserver and getting spammed in return

**Comment by dmitryn** [comment]
screenshot 2018-05-11 14 46 04

This is how it looks like, similar to "Offline" notification

File: src/status_im/transport/inbox.cljs [hunk 1]
```diff
@@ -177,20 +177,19 @@
...
-         wnode    (get-current-wnode-address db)]
+ (fn [{{:keys [web3 network-status] :as db} :db} [_ peers retries]]
+   (let [wnode    (get-current-wnode-address db)]
      (log/info "offline inbox: fetch-peers response" peers)
      (if (registered-peer? peers wnode)
        {::mark-trusted-peer {:web3  web3
                              :wnode wnode}}
        (do
          (log/info "Peer" wnode "is not registered. Retrying fetch peers.")
          (let [delay (if (< retries 3) 300 5000)]
```
Comment (rasom) [inline src/status_im/transport/inbox.cljs]: would be great to extract all magic values to `def`s

File: src/status_im/ui/components/connectivity/view.cljs [hunk 2]
```diff
@@ -31,11 +31,15 @@
...
           (let [pending? (and (:pending @current-chat-contact) (= :chat @view-id))]
-            [react/animated-view {:style (styles/offline-wrapper top offline-opacity window-width pending?)}
+            [react/animated-view {:style (styles/text-wrapper top offline-opacity window-width pending?)}
              [react/view
-              [react/text {:style styles/offline-text}
-               (i18n/label (if @connection-problem?
-                             :t/connection-problem
-                             :t/offline))]]])))})))
+              [react/text {:style    styles/text
+                           :on-press (if @connection-problem?
```
Comment (rasom) [inline src/status_im/ui/components/connectivity/view.cljs]: i suppose 
```clojure
(when @connection-problem?
  #(re-frame/dispatch [:inbox/reconnect]))
```
would work here, no?

**Review by rasom — APPROVED** [review]

File: src/status_im/transport/inbox.cljs (hunk 1 above)
Comment (dmitryn) [reply src/status_im/transport/inbox.cljs]: Extracted

File: src/status_im/ui/components/connectivity/view.cljs (hunk 2 above)
Comment (dmitryn) [reply src/status_im/ui/components/connectivity/view.cljs]: Good point, changed to `when`

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/esmHaa for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/esmHaa">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/rJSADd for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/rJSADd">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/zgYaoD for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/zgYaoD">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/PT4RJQ for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/PT4RJQ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by Serhy** [comment]
Builds https://i.diawi.com/PT4RJQ (iOS) and https://i.diawi.com/zgYaoD (Android) tested.
Confirmed as works great: `Could not connect to mailserver. Tap to reconnect` is shown on Home or chat screen when fails to connect to mailserver. 
Tap upon the label tries to reconnect. And once the connection established - offline inbox messages received. Messages that were failed to be sent - send and delivered by recipients fine.
No errors in logs!



<blockquote><div><strong><a href="https://i.diawi.com/PT4RJQ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/zgYaoD">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/S8WFCx for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/S8WFCx">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/i1Fo4B for branch PR-4220

<blockquote><div><strong><a href="https://i.diawi.com/i1Fo4B">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request merged**